python/lsst/sconsUtils/scripts.py

```diff
@@ -430,49 +508,66 @@ def doc(config="doxygen.conf.in", projectName=None, projectNumber=None, **kw):
-    #                          dictionary; each key must be an entry in swigNameList, and each
-    #                          value a list of additional source files.
-    #  @param ignoreList       List of ignored tests to be passed to tests.Control (note that
-    #                          ignored tests will be built, but not run).
-    #  @param nobuildList      List of tests that should not even be built.
-    #  @param args             A dictionary of program arguments for tests, passed directly
-    #                          to tests.Control.
-    ##
     @staticmethod
     def tests(pyList=None, ccList=None, swigNameList=None, swigSrc=None,
               ignoreList=None, noBuildList=None, pySingles=None,
               args=None):
+        """Convenience function to replace standard tests/SConscript
+        boilerplate.
+
+        With no arguments, will attempt to figure out which files should be
+        run as tests and which are support code (like SWIG modules).
+
+        Python tests will be marked as dependent on the entire ``#python``
+        directory and any SWIG modules built in the tests directory.  This
+        should ensure tests are always run when their results might have
+        changed, but may result in them being re-run more often
+        than necessary.
+
+        Parameters
+        ----------
+        pyList : `list`, optional
+            A sequence of Python tests to run (including .py extensions).
+            Defaults to a ``*.py`` glob of the tests directory, minus any
+            files corresponding to the SWIG modules in swigFileList.
+            An empty list will enable automated test discovery.
+        ccList : `list`, optional
+            A sequence of C++ unit tests to run (including .cc extensions).
+            Defaults to a ``*.cc`` glob of the tests directory, minus any
+            files that end with ``*_wrap.cc`` and files present in swigSrc.
+        swigNameList : `list`, optional
+            A sequence of SWIG modules to build (NOT including .i extensions).
+        swigSrc : `dict`, optional
+            Additional source files to be compiled into SWIG modules, as a
+            dictionary; each key must be an entry in swigNameList, and each
+            value a list of additional source files.
+        ignoreList : `list`, optional
+            List of ignored tests to be passed to
+            `lsst.sconsUtils.tests.Control` (note that ignored tests will be
+            built, but not run).
+        noBuildList : `list`, optional
+            List of tests that should not even be built.
+        pySingles : `list`, optional
+            A sequence of Python tests to run (including .py extensions)
+            as independent single tests. By default this list is empty
+            and all tests are run in a single pytest call.
+            Items specified here will not appear in the default pyList
+            and should not start with "test_" (such that they will not
```

Comment:
Believe it or not, this line is generating this warning:

```
scripts.py:docstring of lsst.sconsUtils.scripts.BasicSConscript.tests:44: WARNING: Unknown target name: "test".
```

(that is, it wants to make `test_` a link.

Luckily it can be fixed by making it a literal:

```rst
``"test_"``
```

Reply:
Thank you!  That warning was bugging me and I couldn't work out what line it was talking about.